src/dotnet/Telemetry.cs

```diff
@@ -31,12 +42,26 @@ public Telemetry()
                 return;
             }
 
+            int sampleRate = ReciprocalSampleRateValue;
+            bool isCITestMachine = Env.GetEnvironmentVariableAsBool(ContinousIntegrationFlag);
```

Comment:
The EnvVar would be set on the CI machines. 

This isn't finalized yet, but was to help further the conversation regarding CI machines.